consensus/core/src/transactions_client.rs

```diff
@@ -20,32 +18,45 @@ const MAX_PENDING_TRANSACTIONS: usize = 2_000;
 #[allow(dead_code)]
 pub(crate) struct TransactionsConsumer {
     tx_receiver: metered_channel::Receiver<Transaction>,
-    max_fetched_per_request: usize,
+    max_fetched_bytes_per_request: u64,
```

Comment:
nit: I think it is also common to use the `fetch` verb for remote fetching. How about just `max_consumed_bytes`?
Alternatively, we can pass in MAX_PENDING_TRANSACTIONS and `max_consumed_bytes` as inputs to `next()`.